app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/blocklist/AppTPBlockListInterceptorApiPluginTest.kt

```diff
@@ -0,0 +1,239 @@
+        testBlockListFeature.atpTdsNextExperimentTest().setRawStoredState(
+            makeExperiment(
+                controlUrl = "controlUrl1",
+                treatmentUrl = "treatmentUrl1",
+            ),
+        )
+        testBlockListFeature.atpTdsNextExperimentAnotherTest().setRawStoredState(
+            makeExperiment(
+                controlUrl = "controlUrl2",
+                treatmentUrl = "treatmentUrl2",
+            ),
+        )
+        checkEndpointIntercept("controlUrl1")
+    }
+
+    @Test
+    fun `when cohort is treatment use treatment URL`() {
+        testBlockListFeature.atpTdsNextExperimentAnotherTest().setRawStoredState(makeExperiment(useTreatment = true))
+        checkEndpointIntercept("treatmentUrl")
+    }
+
+    @Test
+    fun `when cohort is control use control URL`() {
+        testBlockListFeature.atpTdsNextExperimentAnotherTest().setRawStoredState(makeExperiment())
+        checkEndpointIntercept("controlUrl")
+    }
+
+    @Test
+    fun `when feature is for next URL rollout then use next url`() {
+        testBlockListFeature.atpTdsNextExperimentAnotherTest().setRawStoredState(
+            State(
+                remoteEnableState = true,
+                enable = true,
+                settings = configAdapter.toJson(
+                    Config(nextUrl = "nextUrl"),
+                ),
+            ),
+        )
+        checkEndpointIntercept("nextUrl")
+    }
+
+    @Test
+    fun `when no experiments enabled, use default path`() {
+        checkEndpointIntercept(APPTP_TDS_PATH)
+    }
+
+    @Test
+    fun `when feature name doesn't match prefix, it is ignored`() {
+        testBlockListFeature.nonMatchingFeatureName().setRawStoredState(makeExperiment())
+        checkEndpointIntercept(APPTP_TDS_PATH)
+    }
+
+    @Test
+    fun `when annotation not present, even if experiment present, ignore and proceed as normal`() {
+        testBlockListFeature.atpTdsNextExperimentAnotherTest().setRawStoredState(makeExperiment())
+        // even though there's an experiment, we expect the request to not be altered,
+        // as the requesting method doesn't have the annotation
+        checkEndpointIntercept(APPTP_TDS_PATH, requestMethodName = "endpointNotRequiringTds")
+    }
+
+    @Test
+    fun `when experiment request succeeds, doesn't send failure pixel`() {
+        testBlockListFeature.atpTdsNextExperimentAnotherTest().setRawStoredState(makeExperiment())
+        checkEndpointIntercept("controlUrl", expectedResponseCode = 200)
+        verifyNoInteractions(pixel)
+    }
+
+    @Test
+    fun `when experiment request fails, sends failure pixel`() = runTest {
+        testBlockListFeature.atpTdsNextExperimentAnotherTest().setRawStoredState(makeExperiment())
+        checkEndpointIntercept("controlUrl", expectedResponseCode = 400)
+        verify(pixel).fire("experiment_metrics_atpTdsNextExperimentAnotherTest_control", mapOf("code" to "400"))
```

Comment:
Hey @aitorvs

I'm having a bit of trouble getting this test to pass. I've tried adding advanceUntilIdle, runCurrent, and a delay, but the test doesn't seem to be picking up the firing pixel (I can see it in testing on a real device). Any tips?

Another option is to switch this to a regular pixel with the experiment name + group as parameters. Unfortunately, it looks like additional info like appVersion and atb gets added in that case, which we don't necessarily want (in the spirit of minimizing information). Is there a way to turn off those extra params?

Reply:
I've removed the test as it isn't mandatory to have this tested, and I can see the pixel firing on the test device. Would be nice to have the test but if it's hard to get working we can skip.